reactor-core/src/main/java/reactor/core/publisher/UnicastProcessor.java

```diff
@@ -35,10 +35,47 @@
 
 /**
  * A Processor implementation that takes a custom queue and allows
- * only a single subscriber.
+ * only a single subscriber. UnicastProcessor allows demultiplexing of the events which
```

Comment:
Maybe clarify that this is only valid if a custom **Multi-Producer Single-Consumer** `Queue` is provided, **which is not the default**. Otherwise it could break RS if `Publisher`s publish on different threads.